pkg/restapi/issuer/operation/operations.go

```diff
@@ -709,15 +753,23 @@ func (o *Operation) handleRequestPresentation(msg service.DIDCommAction) (interf
 	}), nil
 }
 
-func (o *Operation) generateUserPresentation(handle *AuthorizationCredentialHandle, profile *issuer.ProfileData) (*verifiable.Presentation, error) { // nolint: lll
-	dataReq := &UserDataReq{Token: handle.Token}
+func (o *Operation) createCredential(token string, profile *issuer.ProfileData, authZReq bool) (*verifiable.Credential, error) { // nolint:lll,funlen,gocyclo
+	dataReq := &UserDataReq{Token: token}
 
 	reqBytes, err := json.Marshal(dataReq)
 	if err != nil {
 		return nil, err
 	}
 
-	req, err := http.NewRequest(http.MethodPost, getUserDataURL(profile.URL), bytes.NewBuffer(reqBytes))
+	assuranceCred := false
+	url := getUserDataURL(profile.URL)
+
+	if authZReq && profile.SupportsAssuranceCredential {
```

Comment:
Hard-coding special cases is not sustainable.

Reply:
Created an issue https://github.com/trustbloc/edge-adapter/issues/283